tests/integration/pox/stacking/direct-stacking-extend.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { DevnetNetworkOrchestrator } from "@hirosystems/stacks-devnet-js";
+import { StacksTestnet } from "@stacks/network";
+import { Accounts } from "../../constants";
+import {
+  buildDevnetNetworkOrchestrator,
+  getNetworkIdFromEnv,
+  waitForStacksTransaction,
+} from "../../helpers";
+import {
+  getPoxInfo,
+  waitForNextRewardPhase,
+  readRewardCyclePoxAddressForAddress,
+} from "../helpers";
+import {
+  broadcastStackExtend,
+  broadcastStackSTX,
+} from "../helpers-direct-stacking";
+import { ClarityValue, cvToString, uintCV } from "@stacks/transactions";
+
+describe("testing stack-extend functionality", () => {
+  let orchestrator: DevnetNetworkOrchestrator;
+  let timeline = {
+    epoch_2_0: 100,
+    epoch_2_05: 102,
+    epoch_2_1: 106,
+    pox_2_activation: 109,
+  };
+
+  beforeAll(() => {
+    orchestrator = buildDevnetNetworkOrchestrator(getNetworkIdFromEnv());
+    orchestrator.start();
+  });
+
+  afterAll(() => {
+    orchestrator.terminate();
+  });
+
+  it("stacking and then extending should result in rewards for 3 cycles", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Wait for Stacks genesis block
+    await orchestrator.waitForNextStacksBlock();
+    // Wait for block N+1 where N is the height of the next reward phase
+    await waitForNextRewardPhase(network, orchestrator, 1);
+
+    const blockHeight = timeline.pox_2_activation + 1;
+    const fee = 1000;
+
+    // Alice stacks 50m STX for 1 cycle
+    let response = await broadcastStackSTX(
+      2,
+      network,
+      50_000_000_000_000,
+      Accounts.WALLET_1,
+      blockHeight,
+      1,
+      fee,
+      0
+    );
+    expect(response.error).toBeUndefined();
+
+    // Alice extends stacking for another 2 cycles
+    response = await broadcastStackExtend(
+      network,
+      Accounts.WALLET_1,
+      2,
+      fee,
+      1
+    );
+    expect(response.error).toBeUndefined();
+
+    // Wait for Alice's stacking extension transaction to confirm
+    await waitForStacksTransaction(orchestrator, response.txid);
+
+    // Check rewards for 3 cycles
+    for (let cycle = 1; cycle <= 3; cycle++) {
+
+      let poxInfo = await getPoxInfo(network);
+
+      // Asserts about pox info for better knowledge sharing
+      expect(poxInfo.contract_id).toBe("ST000000000000000000002AMW42H.pox-2");
+      expect(poxInfo.current_cycle.id).toBe(cycle);
+
+      const poxAddrInfo = (await readRewardCyclePoxAddressForAddress(
+        network,
+        cycle+1, // cycle + 1 because we are checking the next cycle, including rewards
+        Accounts.WALLET_1.stxAddress
+      )) as Record<string, ClarityValue>;
+      // IS THIS A BUG? SHOULD BE > 50mm for any of the 3 cycles (increasingly so)
```

Comment:
Why?
The rewards arrive in the btc address?
You can also assert about the locking state of Alice using `expectAccountToBe`

Reply:
not a bug, it was no activated you mentioned.

Reply:
I cannot see the rewards using this:

```typescript
      const poxAddrInfo = (await readRewardCyclePoxAddressForAddress(
        network,
        cycle+1, // cycle + 1 because we are checking the next cycle, including rewards
        Accounts.WALLET_1.stxAddress
      )) as Record<string, ClarityValue>;
      expect(poxAddrInfo?.["total-ustx"]).toEqual(uintCV(80_000_000_000_000));
```

Reply:
No, only the amounts used to calculate the slots.

Reply:
okey, is working.